lib/public/views/Logs/FilePreview/filePreviewComponent.js

```diff
@@ -83,93 +97,113 @@ export const showFilePreview = (filePreviewModel, attachment) => {
      * @returns {vnode} the preview popover
      */
     const previewPopover = (content) => popover(
-        fileName('text-decoration: underline; cursor: pointer;'),
-        h('', {
-            style: 'margin: 1%; max-height: 99%; overflow: hidden; display: flex; flex-direction: column;',
-        }, [
-            h('.f3', {}, attachment.name),
-            h('', {
-                style: 'overflow: auto;',
-            }, content),
-        ]),
+        previewButton,
+        popoverWindow(content),
         {
             ...PopoverTriggerPreConfiguration.click,
             anchor: PopoverAnchors.LEFT_MIDDLE,
         },
     );
 
-    const previewFileStyle = 'margin: 0; width: 100%;';
+    /**
+     * Checks if the file is a blob or not, and prepares the file to show
+     *
+     * @param {Method} awaitMethod actions to take after onload, or fetching
+     * @param {Method} readMethod reader method to execute (readAsText, readAsDataUrl, ...)
+     * @returns {void}
+     */
+    const readFile = (awaitMethod, readMethod) => {
```

Comment:
All this should be in the model. I would say that you should have a file preview remote data (a concrete class per file type) and you pass this to a component to display file preview (a component per file type).
If you want to make things a bit simpler, you can use a `StatefulComponent`.